doit/dependency.py

```diff
@@ -361,6 +338,96 @@ def remove_all(self):
         self._conn.execute('delete from doit')
 
 
+class BaseChecker(object):
+    """Base Checker that must be inherited."""
+
+    def __init__(self, backend):
```

Comment:
I did it this way because in the md5 checker there is a call to get the current state.
So instead I know pass the current state to `get_state`.